hs_core/hydroshare/resource.py

```diff
@@ -1121,7 +1122,10 @@ def publish_resource(user, pk):
                'url': get_activated_doi(resource.doi)}
     resource.metadata.create_element('Identifier', **md_args)
 
+    # async regenerate the bag to ensure that the bag is complete at time of publication.
+    # it serves as a reference in case anything happens with the filesystem
```

Comment:
Are we assuming the bag won't be regenerated once the resource is published?

Reply:
Good point @pkdash we don't prevent owners from editing metadata (with the exception of Title, Authors, and Content Files) so we really can't assume that the bag created at the time of publication won't be overwritten with modifications.

@martinseul I was forgetting that metadata (and thus the bag) can be edited _after publication_, so I think we don't really gain anything by forcing bag overwrite at the time of publication.